jaeger-tracerresolver/src/main/java/io/jaegertracing/tracerresolver/internal/JaegerTracerFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2018, The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.jaegertracing.tracerresolver.internal;
+
+import io.jaegertracing.Configuration;
+import io.jaegertracing.internal.JaegerTracer;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.tracerresolver.TracerFactory;
+
+public class JaegerTracerFactory implements TracerFactory {
+  @Override
+  public Tracer getTracer() {
+    return getJaegerTracer();
+  }
+
+  public JaegerTracer getJaegerTracer() {
+    return Configuration.fromEnv().getTracer();
+  }
```

Comment:
I suggest dropping this method

Reply:
And leave the old resolver as it is, right?

Reply:
Don't know about the resolver, no comment. But this method should just be folded into the one above, it doesn't add anything useful.

Reply:
I simplified this a bit. As the resolver was also calling the `JaegerTracerFactory#getJaegerTracer()` method, I just changed the return object from `getTracer()` to be `JaegerTracer`. This way, the old resolver can still call it, ensuring that both the resolver and the factory yields similar tracers.

Reply:
Cool, there's a new "Resolve conversation" button here :)